### PR TITLE
CHI-1040: Split Case.tsx

### PR DESCRIPTION
**Pull request by stephenhand**
Primary reviewer: @murilovmachado 

## Description

* Splits `Case.tsx` into a 'parent component' which doesn't render anything and just handles subrouting and holds shared functionality, and `CaseHome.tsx` which renders an individual case's landing or 'home' page.
* Currently updating the connected case is handled in the parent which delegates functions to `CaseHome` as properties, in line with how `CasePrintView` works, it also reuses the `CaseDetails` structure that `CasePrintView` uses
* Tests are also split. Case.tsx could use some additional coverage.
* Commented out 'Cancel' test reimplemented
* Case.test.js migrated to TS

### Checklist
- [ X ] Corresponding issue has been opened
- [ X ] New tests added
- [ NO ] Strings are localized
- [ X ] Tested for chat contacts
- [ NO ] Tested for call contacts

### Related Issues
* Implements CHI-1040

### Verification steps

* Regression test case functionality (in new cases and from the list).

### CODE REVIEW

File: plugin-hrm-form/src/components/case/Case.tsx [hunk 1]
```diff
@@ -611,150 +441,21 @@ const Case: React.FC<Props> = props => {
...
+          handleClose={handleClose}
+          handleCancelNewCaseAndClose={handleCancelNewCaseAndClose}
+          handleUpdate={handleUpdate}
+          handleSaveAndEnd={handleSaveAndEnd}
+          onInfoChange={onInfoChange}
+          onStatusChange={onStatusChange}
+          isCreating={isCreating}
+          isEdited={Boolean(caseHasBeenEdited)}
+          can={can}
+        />
```
Comment (murilovmachado) [inline plugin-hrm-form/src/components/case/Case.tsx:458]: Great!! We've been postponing this refactoring for so long 😅

**Review by murilovmachado — APPROVED** [review]
I've mostly glanced through the most important parts here, and run the code locally.
Seems good to me!

**Review by GPaoloni — APPROVED** [review]
Code looks amazing, thank you very much!!

File: plugin-hrm-form/src/components/case/Timeline.tsx [hunk 2]
```diff
@@ -31,18 +31,18 @@ import { NewCaseSubroutes, AppRoutesWithCase } from '../../states/routing/types'
 
 type OwnProps = {
   timelineActivities: Activity[];
+  contacts: any[];
```
[the remark this answers is not in the document]
Comment (stephenhand) [reply plugin-hrm-form/src/components/case/Timeline.tsx:34]: The property on `contacts` property on `CaseDetails` was `any[]` so I propagated that through. I could have changed the CaseDetails property type, but I wanted to draw the line on the refactoring somewhere 😄

**Pull request merged**